src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/CreateUserProfileFuncTest.java

```diff
@@ -48,15 +46,66 @@ public void should_create_user_profile_and_verify_successfully() throws Exceptio
     }
 
     @Test
-    public void should_create_user_profile_for_duplicate_idam_user_and_verify_successfully() throws Exception {
+    public void should_create_user_profile_for_duplicate_idam_user_and_verify_successfully_for_prd_roles() throws Exception {
 
-        String email = idamClient.createUser("pui-user-manager");
+        List<String> xuiuRoles = new ArrayList();
+        xuiuRoles.add("pui-user-manager");
+        xuiuRoles.add("pui-case-manager");
 
+        //create user with "pui-user-manager" role in SIDAM
+        List<String> sidamRoles = new ArrayList<>();
+        sidamRoles.add("pui-user-manager");
+        String email = idamClient.createUser(sidamRoles);
+
+        //create User profile with same email to get 409 scenario
+        CreateUserProfileData data = createUserProfileData();
+        data.setRoles(xuiuRoles);
+        data.setEmail(email);
+        CreateUserProfileResponse duplicateUserResource = createUserProfile(data, HttpStatus.CREATED);
+        verifyCreateUserProfile(duplicateUserResource);
+
+        //get user by getUserById to check new roles got added in SIDAM
+        //should have 2 roles
+        String userId = duplicateUserResource.getIdamId();
+        GetUserProfileWithRolesResponse resource =
+                testRequestHandler.sendGet(
+                        requestUri + "/" + userId + "/roles",
+                        GetUserProfileWithRolesResponse.class);
+
+        assertThat(resource.getRoles()).contains("pui-case-manager");
+        assertThat(resource.getRoles()).contains("pui-user-manager");
+
+    }
+
+    @Test
+    public void should_create_user_profile_for_duplicate_idam_user_and_verify_roles_updated_successfully_for_user_having_citizen_role() throws Exception {
+
+        //create user with citizen role in SIDAM
+        List<String> roles = new ArrayList<>();
+        roles.add("citizen");
+        roles.add("pui-case-manager");
+        String email = idamClient.createUser(roles);
+
+        //create user profile in UP with PRD-ADMIN token for above user with same email with "pui-user-manager" roles
         CreateUserProfileData data = createUserProfileData();
         data.setEmail(email);
+        List<String> xuiRoles = new ArrayList();
+        xuiRoles.add("pui-user-manager");
+        xuiRoles.add("pui-case-manager");
+        data.setRoles(xuiRoles);
         CreateUserProfileResponse duplicateUserResource = createUserProfile(data, HttpStatus.CREATED);
         verifyCreateUserProfile(duplicateUserResource);
 
+        //get user by getUserById to check new roles got added in SIDAM
+        String userId = duplicateUserResource.getIdamId();
+        GetUserProfileWithRolesResponse resource =
+                testRequestHandler.sendGet(
+                        requestUri + "/" + userId + "/roles",
+                        GetUserProfileWithRolesResponse.class);
+
+        assertThat(resource.getRoles()).contains("citizen");
```

Comment:
contains check put only one condition and add all the roles with coma separated

Reply:
since its test case code and not functional, its ok to check only 2 roles separately since it will not cause any performance issue.